src/Mod/TechDraw/App/DrawViewDimension.cpp

```diff
@@ -574,107 +574,130 @@ std::string  DrawViewDimension::getFormatedValue(int partial)
+        // if useDecimals(), then formatSpecifier = global decimals, otherwise it is %.2f
+        QLocale loc;
+        double userVal;
+        bool checkDecimals = true;
+        if (showUnits() || (Type.isValue("Angle")) || (Type.isValue("Angle3Pt"))) {
+            formattedValue = qUserString; // result value + unit (not necessarily base unit!)
+            // remove unit
+            formattedValue.remove(rxUnits);
+            // to number
+            userVal = loc.toDouble(formattedValue);
+            if (userVal >= 1.0)
+                // we can assure we didn't make an error > 10% via getUserString()
+                checkDecimals = false;
+        }
+        if (checkDecimals){
+            // get value in the base unit with default decimals
+            // for the conversion we use the same method as in DlgUnitsCalculator::valueChanged
+            // at first remove space from unit string
+            QRegExp space(QString::fromUtf8("\\s"));
+            QString UnitString = qUserStringUnits;
+            UnitString.remove(space);
+            // if the unit is a metric lenght unit, nothing has to be converted
+            if ( UnitString == QString::fromUtf8("\xC2\xB5m") || UnitString == QString::fromLatin1("mm")
+                || UnitString == QString::fromLatin1("nm") || UnitString == QString::fromLatin1("m")
+                || UnitString == QString::fromLatin1("km") ) {
+                userVal = asQuantity.getValue();
             }
-        } else {
-            if (showUnits() || (Type.isValue("Angle")) || (Type.isValue("Angle3Pt"))) {
-                repl = formattedValue + qUserStringUnits;
-            } else {
-                repl = formattedValue;
+            else {
+                double convertValue = Base::Quantity::parse(QString::fromLatin1("1") + UnitString).getValue();
+                // we got now e.g. for "1 in" the value '25.4' because 1 in = 25.4 mm
+                // the result is now just val / convertValue because val is always in the base unit
+                userVal = asQuantity.getValue() / convertValue;
             }
         }
+        // we reformat the value
+#if QT_VERSION >= 0x050000
+        formattedValue = QString::asprintf(Base::Tools::toStdString(formatSpecifier).c_str(), userVal);
+#else
+        QString qs2;
+        formattedValue = qs2.sprintf(Base::Tools::toStdString(formatSpecifier).c_str(), userVal);
+#endif
+        // if abs(1 - userVal / formattedValue) > 0.1 we know that we make an error greater than 10%
+        // then we need more digits
+        if (userVal > 0.0 && abs(1 - formattedValue.toDouble() / userVal) > 0.1) {
+            int i = 1;
+            do { // increase decimals step by step until error is < 10 %
+                formattedValue = QLocale().toString(userVal, 'f', i);
+                ++i;
+            } while (abs(1 - loc.toDouble(formattedValue) / userVal) > 0.1);

```

Comment:
>  If the user has set the Format Spec to %.1f, it might be by accident or by intent,

Setting to one decimal for a value < 0.1 is a mistake and we discussed in the forum that outputting "0.0" is a no-go. Such a drawing is useless and violates the rules for a technical drawing. Such things occur either if you accidentally change the unit system to something unsuitable for an existing drawing or you make the mistake of setting a global decimal number that is too low for your use case. In both cases you see your mistake and can either change the unit system or increase the number of decimals.
A typical use case are, however, real-life drawings where I only need 2 decimals (cannot be milled more precisely) and only for very special fine ones I need 3 (the 3D printer has a resolution of 25 µm). So also in the past I only changed this special dimension to have 3 decimals to not pollute the whole drawing with zeros in the dozen other dimensions.

Reply:
Yes, what you said is true; and that's what I like in your PR; i.e. it fixes the real-life use case.  I have no objection here, if others accept this, too.